### DISCUSSION TITLE
htmx@2 veryfyPath issue

### DISCUSSION THREAD

**Discussion by imbolc** [bigskysoftware/htmx#2546] [Q&A] [answered]
Is [verifyPath](https://github.com/bigskysoftware/htmx/blob/57595bc0390b6ed98ada45a5d27156322e308757/src/htmx.js#L3028) is going to be stabilized in v2? What's its purpose? Is there some security issues in requesting wrong paths?

I'm trying to update [htmx-template](https://github.com/kgscialdone/htmx-template/blob/master/htmx-template.js) extension to work with htmx@2. The issue is that it assumes urls are started with a not standard prefix: `hx-get="template:lorem"`. So `verifyPath` just throws an error and the extension never gets the original event.

If `verifyPath` is going to be stabilized what would be your recommendation on the extension. It works if I change prefix to a correct url e.g. `hx-get="hx-template#lorem"`, but I wonder if there's a better way?

**Comment by kgscialdone** [comment, marked as answer]
`verifyPath` hasn't changed. This is caused by the default value of the `selfRequestsOnly` config parameter changing from `false` to `true` in 2.x. It isn't a bug, rather the fact that the extension works in the first place is a "bug" and would break if the config is changed even on 1.x.

**Reply by imbolc** [reply]
Thank you :)

**Reply by cryp2one** [reply]
please update in Document https://htmx.org/reference/
htmx.config.selfRequestsOnly  defaults to false, if set to true will only allow AJAX requests to the same domain as the current document

**Reply by imbolc** [reply]
I ended up using the changes I described in the question: https://gist.github.com/imbolc/76b3a05fa24a7f32fccfbd7d9d0c5d41
